engines/supernova/metaengine.cpp

```diff
@@ -201,6 +207,217 @@ SaveStateDescriptor SupernovaMetaEngine::querySaveMetaInfos(const char *target,
+	act->setCustomEngineActionEvent(kActionPgUp);
+	act->addDefaultInputMapping("PAGEUP");
+	act->addDefaultInputMapping("JOY_RIGHT_SHOULDER");
+	textReaderKeymap->addAction(act);
+
+	act = new Common::Action("PGDOWN", _("Page down"));
+	act->setCustomEngineActionEvent(kActionPgDown);
+	act->addDefaultInputMapping("PAGEDOWN");
+	act->addDefaultInputMapping("JOY_LEFT_SHOULDER");
+	textReaderKeymap->addAction(act);
+
+	act = new Common::Action("EXIT", _("Exit"));
+	act->setCustomEngineActionEvent(kActionExit);
+	act->addDefaultInputMapping("ESCAPE");
+	act->addDefaultInputMapping("JOY_B");
+	textReaderKeymap->addAction(act);
+
+	act = new Common::Action("SELECT", _("Select"));
+	act->setLeftClickEvent();
+	act->addDefaultInputMapping("MOUSE_LEFT");
+	act->addDefaultInputMapping("JOY_A");
+	menuKeymap->addAction(act);
+
+	act = new Common::Action("SKIP", _("Skip cutscene"));
+	act->setCustomEngineActionEvent(kActionSkip);
+	act->addDefaultInputMapping("ESCAPE");
+	act->addDefaultInputMapping("JOY_B");
+	cutsceneKeymap->addAction(act);
+
+	act = new Common::Action("SKPL", _("Skip line"));
+	act->setCustomEngineActionEvent(kActionSkipLine);
+	act->addDefaultInputMapping("SPACE");
+	act->addDefaultInputMapping("JOY_A");
+	cutsceneKeymap->addAction(act);
+
+	act = new Common::Action("GO", _("Go"));
+	act->setCustomEngineActionEvent(kActionGo);
+	act->addDefaultInputMapping("1");
+	improvedKeymap->addAction(act);
+
+	act = new Common::Action("LOOK", _("Look"));
+	act->setCustomEngineActionEvent(kActionLook);
+	act->addDefaultInputMapping("2");
+	improvedKeymap->addAction(act);
+
+	act = new Common::Action("TAKE", _("Take"));
+	act->setCustomEngineActionEvent(kActionTake);
+	act->addDefaultInputMapping("3");
+	improvedKeymap->addAction(act);
+
+	act = new Common::Action("OPEN", _("Open"));
+	act->setCustomEngineActionEvent(kActionOpen);
+	act->addDefaultInputMapping("4");
+	improvedKeymap->addAction(act);
+
+	act = new Common::Action("CLOSE", _("Close"));
+	act->setCustomEngineActionEvent(kActionClose);
+	act->addDefaultInputMapping("5");
+	improvedKeymap->addAction(act);
+
+	act = new Common::Action("PUSH", _("Push"));
+	act->setCustomEngineActionEvent(kActionPush);
+	act->addDefaultInputMapping("6");
+	improvedKeymap->addAction(act);
+
+	act = new Common::Action("PULL", _("Pull"));
+	act->setCustomEngineActionEvent(kActionPull);
+	act->addDefaultInputMapping("7");
+	improvedKeymap->addAction(act);
+
+	act = new Common::Action("USE", _("Use"));
+	act->setCustomEngineActionEvent(kActionUse);
+	act->addDefaultInputMapping("8");
+	improvedKeymap->addAction(act);
+
+	act = new Common::Action("TALK", _("Talk"));
+	act->setCustomEngineActionEvent(kActionTalk);
+	act->addDefaultInputMapping("9");
+	improvedKeymap->addAction(act);
+
+	act = new Common::Action("GIVE", _("Give"));
+	act->setCustomEngineActionEvent(kActionGive);
+	act->addDefaultInputMapping("0");
+	improvedKeymap->addAction(act);
+
+	if(gameId == "msn1") {
```

Comment:
```suggestion
	if (gameId == "msn1") {
```
See https://wiki.scummvm.org/index.php?title=Code_Formatting_Conventions#Whitespaces